packages/gatsby-plugin-mdx/src/gatsby-node.ts

```diff
@@ -76,16 +121,30 @@ export const preprocessSource: GatsbyNode["preprocessSource"] = async (
     return undefined
   }
 
-  const code = await compile(contents, mdxOptions)
+  const code = await compileMDX(
+    contents,
+    {
```

Comment:
What are these two empty objects doing?

Reply:
They are mocked MDX and File nodes, as these are required by some gatsby-remark-* plugins to properly parse and render the MDX.

Here they are mocked, as they are not existing yet in the`preprocessSource` step (correct me if I am wrong)

As this is mostly done to locate static GraphQL queries, I wonder if we should just skip all remark/rehype/gatsby-remark-* plugins in this step. That way we'd not need to pass any data down the pipeline here, probably also save some processing time.